### PR TITLE
Decrease crap level (task #8430)

### PR DESCRIPTION
**Pull request by mitsios**
Added a testcase file for FilterQuery among with test scenarios

### CODE REVIEW

**Comment by mamchenkov** [comment]
@mitsios @softius @georgeconstantinou 

I think we shouldn't test private methods directly.  Instead, we should find a way to test their execution via a public method.

File: tests/bootstrap.php [hunk 1]
```diff
@@ -43,6 +43,7 @@
 define('CAKE', CORE_PATH . 'src' . DS);
 
 require ROOT . '/vendor/autoload.php';
+require __DIR__ . '/Class/UsefulMethods.php';
```
Comment (softius) [inline tests/bootstrap.php]: @mitsios please use a namespace to avoid including manually the file

File: tests/Class/UsefulMethods.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?php
+use RolesCapabilities\FilterQuery;
+
+/**
+ * Class for useful methods in regards testing
+ */
+class UsefulMethods
```
Comment (softius) [inline tests/Class/UsefulMethods.php]: @mitsios can we change the name to something like `Helper` or `Utility`?

**Comment by mamchenkov** [comment]
@mitsios @georgeconstantinou @softius 

1. 'Useful Methods' are often called 'Utilities' :) 
2. We can move these into the `cakephp-utils/src/TestSuite/JsonIntegrationTestCase.php` or similar, so that it can be reused by other plugins as well.

**Pull request merged**